### DISCUSSION TITLE
Virtual Screening Enrichment via rdkit.ML.Scoring.Scoring module and rdkit.ML.EnrichPlot module?

### DISCUSSION THREAD

**Discussion by DKchemistry** [rdkit/rdkit#5545] [Q&A]
Hi all, 

I'd like to make plots like these:
 
![image](https://user-images.githubusercontent.com/81147591/189119951-2064a181-3dae-4f96-ab64-8369e2520156.png)
(source: [Deep Docking: A Deep Learning Platform for Augmentation of Structure Based Drug Discovery](https://pubs.acs.org/doi/10.1021/acscentsci.0c00229), Fig 6A)

As well as other metrics and figures used to evaluate enrichment in virtual screening. I have been struggling to find software that will do this. The figure above was generated using OpenEye scripts that I do not have access to. I have analogous Schrodinger scripts that will give integer metrics (EF, ROCAUC, AAUC, BEDROC, etc), but the plotting functions do not work. 

I found these modules in RDKit: [rdkit.ML.Scoring.Scoring](http://rdkit.org/docs/source/rdkit.ML.Scoring.Scoring.html) and [rdkit.ML.EnrichPlot](http://rdkit.org/docs/source/rdkit.ML.EnrichPlot.html) and thought maybe they could do this? But perhaps the EnrichPlot is for something else entirely. 

I have tried looking at the RDKit book, cookbook, and google for examples of how to use these, but haven't found anything. 

The schrodinger versions asked for two required inputs: 

1) results.csv = an ordered scores file containing an ID and score.
2) actives.txt = the IDs of known actives

For something like: 

rdkit.ML.Scoring.Scoring.CalcBEDROC(scores, col, alpha)

I can see what is being asked for in the documentation, but I am not familiar with how to generate an appropriate data structure? Apologies for the naivety, still very much a programming beginner. 

If I have a data file like so, 

mol, docking_score, activity_status
id1, -10, true 
id2, -9, false
id3, -8, true
id4, -7, true 

How do I supply it to rdkit? I'm familiar with basic pandas/numpy data transforms, but not sure what the function wants.

**Comment by gedeck** [comment]
It looks like you want to create a ROC curve. Instead of using RDKit, I suggest you look at this example from scikit-learn:
https://scikit-learn.org/stable/auto_examples/model_selection/plot_det.html

Here is another blog post that demonstrates the use of the scikit-learn function https://scikit-learn.org/stable/auto_examples/model_selection/plot_det.html

The Scoring code is quite outdated and if you only want the ROC curves, I wouldn't use it. However, it contains a few other metrics that might be useful. So, if you still want to use the Scoring code, It takes a list of lists of results and the column that indicates if something is active or not. The relevant line is here: 
https://github.com/rdkit/rdkit/blob/383af8895c55f5d0bb41430bc45abc838954eaf5/rdkit/ML/Scoring/Scoring.py#L32
```
  for i in range(numMol):
    if scores[i][col]:
      numActives += 1
    else:
      numInactives += 1
    TPR[i] = numActives  # TP
    FPR[i] = numInactives  # FP
```
With your data file, once you convert it to e.g.
```
data = [
    ['id1', -10, True],
    ['id2', -9, False],
    ...
]
```
you will need to sort it by the score values `sorted(data, key=lambda x: x[1])`
you should be able to call it with 
```
rdkit.ML.Scoring.Scoring.CalcROC(data, 2)
```
It's been a long time that I've used that, so no promise that this works.
